src/AppInstallerRepositoryCore/ISource.h

```diff
@@ -88,6 +88,12 @@ namespace AppInstaller::Repository
         // Set caller.
         virtual void SetCaller(std::string) {}
 
+        // Require the source to be explicitly declared. Source must be removed for this to be reset.
+        virtual void SetExplicit() {}

```

Comment:
I'm not seeing the reason for `ISourceReference` to need these methods yet.  The data is stored in the details anyway, so why is that not the entry point for changing it?